credentials/credentials.go

```diff
@@ -50,6 +50,23 @@ type PerRPCCredentials interface {
 	RequireTransportSecurity() bool
 }
 
+// SecurityLevel defines the protection level on an established channel.
+type SecurityLevel int
+
+const (
+	// SecurityNone indicates a channel is a plaintext channel that does not provide any protection.
+	SecurityNone SecurityLevel = iota
+	// IntegrityOnly indicates a channel only provides integrity protection.
+	IntegrityOnly
+	// PrivacyAndIntegrity indicates a channel provides both privacy and integrity protection.
+	PrivacyAndIntegrity
+)
+
+// CommonAuthInfo contains authenticated information common to AuthInfo implementations.
+type CommonAuthInfo struct {
+	Level SecurityLevel
```

Comment:
Nit: `SecurityLevel SecurityLevel`.  IMO "Level" by itself is not distinct enough.